script/vsts/platforms/templates/preparation.yml

```
@@ -20,9 +20,17 @@ steps:
     condition: eq(variables['Agent.OS'], 'Linux')
 
   # Common
+  - bash: |
+      if [ "${BUILD_ARCH}" = "x86" ]; then \
+        echo "##vso[task.setvariable variable=SHOULD_FORCE_32_BIT_NODE]true"
+      fi
+    displayName: Set SHOULD_FORCE_32_BIT_NODE based on BUILD_ARCH
+    # Templates evaluate too early for the matrix variable "BUILD_ARCH" to be available. Scripts evaluate at runtime, so "BUILD_ARCH" will be available.
+
   - task: NodeTool@0
     inputs:
       versionSpec: 12.16.3
+      force32bit: $(SHOULD_FORCE_32_BIT_NODE)
```

Why not just set when it is needed? When I wrote these scripts, the reason I used the PowerShell syntax to set the environment variables was because `Agent.OS` was not available in the following simpler syntax. 

But for BUILD_ARCH that should not be the case.
```suggestion
      ${{ if eq(variables['BUILD_ARCH'], 'x86') }}:
      		force32bit: true
```

---

If I commit this change, you will see that both  of the Windows builds will install 64-bit Node.

The code you suggest here was my first instinct, and it does not work.

I applied this suggested diff on a branch and ran CI. See that the "Windows x86" build installs 64-bit Node:
https://dev.azure.com/DeeDeeG/b/_build/results?buildId=857&view=logs&j=e4115def-01ab-568c-a31a-bc15166ffabd&t=12c9d5a1-842e-5bcb-4a43-64a451213953

---

I like your [first alternate design](https://github.com/DeeDeeG/atom/commit/8c046786938dd226722a8870944ddf4d552d5d4d). It is much simpler. Could you use that and instead rename `SHOULD_FORCE_32_BIT_NODE` to `IsWinX86`?

This new variable can be handy in other places as well.

---

Note to Atom's maintainers: If you have an opinion on which approach is best, I'd appreciate hearing from you on this.

- This pull request as it is now
- Alternate Design 1 (as mentioned in the body text of the pull request)
- Alternate Design 2

In any of these cases, the variable named `SHOULD_FORCE_32_BIT_NODE` could be renamed `IsWinX86` instead, if that's nicer. (As suggested by @aminya).

---

I prefer the approach in this PR. I don't see anything problematic with it. As for the variable name `SHOULD_FORCE_32_BIT_NODE` It makes sense to rename it to `IsWinX86`.

---

There is nothing wrong with it. The alternate design 1 is more succinct. 

Similarly, in [this PR](https://github.com/atom/atom/pull/21325/files#diff-f0efa4d06560f5cba2917290fef9cad255405df2775d4a29cc79f8f0bdf82e3cR25), I introduced `IsNightlyBranch`, and I used it every where.


I'll let the final decision up to you. To me that seems simpler.

---

@sadick254 thanks, I'll update the PR.

---

@aminya For the record "all the unnecessary code" is one powershell script spanning three lines. (If not for nice formatting, the script could easily be one line -- if not for spacing out the `{}` brackets of the `if` statement.) With a line of metadata to have a human-readable name in the CI. And one or two lines of comments.

This pull request actually removes more lines than it adds.

When a pull request is this close to optimal, and the different options are so trivially similar, it causes a lot of communication and technical review to have this many comments back and forth. For very little technical difference. It's not worth this much discussion or debate, in my opinion, in order to shuffle one or two variables around.

When a PR is "good enough," I think it's polite to just leave it be. As I have said elsewhere, we all have the chance to contribute in our own style when we contribute. If there are disagreements, official maintainers can decide.

---

Ok. That's fine. It's always possible to update the code in follow up PRs.